crates/cxx-qt-lib/src/core/qtlogging.cpp

```diff
@@ -0,0 +1,48 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qtlogging.h"
+
+#include "../assertion_utils.h"
+
+// QMessageLogContext has three "const char*" members for line, category, etc
+// https://codebrowser.dev/qt5/qtbase/src/corelib/global/qlogging.h.html#QMessageLogContext
+assert_alignment_and_size(QMessageLogContext, alignof(intptr_t), sizeof(intptr_t) * 4);
+
+static_assert(::std::is_trivially_copyable<QMessageLogContext>::value,
```

Comment:
is it actually trivially_copyable ? types that have that don't need `rust::IsRelocatable` definition.


Normally you either have to check all of these, if copy is false then you need to impl `Clone` manually and if deconstructible is failse then you need to manually implement `Drop` (which tends to be the case for these types.
```cpp
// in the header have the isrelocatable
rust::IsRelocatable<T> ...

// in the source
static_assert(!::std::is_trivially_copy_assignable<T>::value);
static_assert(!::std::is_trivially_copy_constructible<T>::value);
static_assert(!::std::is_trivially_destructible<T>::value);

static_assert(QTypeInfo<T>::isRelocatable);
```

or if the type is actually trivial then you only have the following and then derive the rest

```cpp
static_assert(::std::is_trivially_copyable<T>::value);
```

As this type uses pointers internally it suggests it's probably the former (or is it actually trivially copyable as it's just `const char*` ?) but it looks like the type doesn't have any custom deconstructors or constructors so you probably don't implement Clone and Drop, fun :-)

Reply:
I think the type is trivial, although you suggest to remove IsRelocatable in that case (IIUC) the CXX compilation fails with:

```
  cargo:warning=/home/josh/work/cxx-qt/target/debug/build/cxx-qt-lib-9ea1616c001b64fd/out/cxx-qt-gen/src/qtlogging.cxx.cpp:62:50: error: static assertion failed: type QMessageLogContext should be trivially move constructible and trivially destructible in C++ to be used as a return value of `qmessagelogcontext_default` or non-pinned mutable reference in signature of `set_line`, `set_file`, `set_function` in Rust
  cargo:warning=   62 |     ::rust::IsRelocatable<::QMessageLogContext>::value,
```